### PR TITLE
Migrate highlight.js API Call Usage in TextualBody.tsx

### PR DESCRIPTION
**Pull request by mustafa-kapadia1483**
## Checklist

-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

Type: Task
Updated highlight usage to use highlight(code, options) instead as highlight(lang, code, ...args) has been deprecated
fixes [Update highlight.js usage in-code #22233](https://github.com/vector-im/element-web/issues/22233)

Signed-off-by: Mustafa Kapadia <mustafakapadia76@gmail.com>

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/components/views/messages/TextualBody.tsx [hunk 1]
```diff
@@ -238,18 +238,19 @@ export default class TextualBody extends React.Component<IBodyProps, IState> {
...
+                    codeTextContent.length +
                     " > " +
                     MAX_HIGHLIGHT_LENGTH +
                     "): not highlighting",
             );
             return;
         }
 
-        let advertisedLang;
+        let advertisedLang = "";
```
Comment (dbkr) [inline src/components/views/messages/TextualBody.tsx:253]: I'm not sure I understand why this change is here: we're using it as a condition on the 'if' block so it being undefined or a string feels more correct?

File: src/components/views/messages/TextualBody.tsx [hunk 2]
```diff
@@ -238,18 +238,19 @@ export default class TextualBody extends React.Component<IBodyProps, IState> {
     }
 
     private highlightCode(code: HTMLElement): void {
-        if (code.textContent.length > MAX_HIGHLIGHT_LENGTH) {
+        const codeTextContent = code.textContent ?? "";
```
Comment (dbkr) [inline src/components/views/messages/TextualBody.tsx:241]: I assume this change is to make the strict mode checker pass? If we are checking, we may as well just early exit if the text content is empty: we can't syntax highlight the empty string.

**Review by dbkr — CHANGES_REQUESTED** [review]

**Comment by dbkr** [comment]
Also, I don't think you need a 'Notes' tag in this PR: it's not a user-facing change so it doesn't need to be in the changelog (the explanation is useful though, so I'd keep it in the description, just not as a changelog note).

**Comment by mustafa-kapadia1483** [comment]
Thank You @dbkr for reviewing the changes, I have made the changes as per the review

File: src/components/views/messages/TextualBody.tsx [hunk 3]
```diff
@@ -238,10 +238,15 @@ export default class TextualBody extends React.Component<IBodyProps, IState> {
     }
 
     private highlightCode(code: HTMLElement): void {
-        if (code.textContent.length > MAX_HIGHLIGHT_LENGTH) {
+        const codeTextContent = code.textContent;
+        if (!codeTextContent) {
+            console.log("Code block is empty");
```
Comment (dbkr) [inline src/components/views/messages/TextualBody.tsx:243]: We use the loglevel library for logging, so no console.log statements, although I don't think a log line is necessary here - it could get quite noisy.

**Review by dbkr — CHANGES_REQUESTED** [review]

File: src/components/views/messages/TextualBody.tsx (hunk 3 above)
Comment (mustafa-kapadia1483) [reply src/components/views/messages/TextualBody.tsx:243]: there was a console.log when codeTextContent.length > MAX_HIGHLIGHT_LENGTH hence I thought same process should be replicated when exiting early for empty string / null

ref:
```javascript 
if (codeTextContent.length > MAX_HIGHLIGHT_LENGTH) {
    console.log(
        "Code block is bigger than highlight limit (" +
            codeTextContent.length +
            " > " +
            MAX_HIGHLIGHT_LENGTH +
            "): not highlighting",
    );
    return;
}
```

**Comment by t3chguy** [comment]
Duplicate of https://github.com/matrix-org/matrix-react-sdk/pull/9923

**Comment by mustafa-kapadia1483** [comment]
> Duplicate of #9923

@t3chguy, this PR was fixing the static Static Analysis / Typescript Strict Error Checker (--strict --noImplicitAny) error in https://github.com/matrix-org/matrix-react-sdk/pull/9923
